server/workflow/workflow_server_test.go

```diff
@@ -638,7 +638,7 @@ func getWorkflowServer() (workflowpkg.WorkflowServiceServer, context.Context) {
 	if err = wfStore.Add(&wfObj5); err != nil {
 		panic(err)
 	}
-	server := NewWorkflowServer(instanceIdSvc, offloadNodeStatusRepo, archivedRepo, wfClientset, wfStore, wfStore)
+	server := NewWorkflowServer(instanceIdSvc, offloadNodeStatusRepo, archivedRepo, wfClientset, wfStore, wfStore, nil)
```

Comment:
oh, I think I changed it by accident, and I changed it back the `metav1.NamespaceAll`

> With the new && namespace != nil check, this now takes a different path (but still passes apparently?)

This doesn't really matter regarding the test because we don't [start the reflector to perform list and watch](https://github.com/argoproj/argo-workflows/blob/main/server/workflow/workflow_server.go#L86-L88) in the test and we just [manually add objects to the reflector/store](https://github.com/argoproj/argo-workflows/blob/main/server/workflow/workflow_server_test.go#L631-L640).

Reply:
But the reflector is [set](https://github.com/argoproj/argo-workflows/blob/dc6a18dc545bbf35103ffddb81a014eefcbb403d/server/workflow/workflow_server.go#L79) since the store is not `nil` in this case, so it would start wouldn't it? It just isn't used actively in this test

Reply:
the [`Run` method](https://github.com/argoproj/argo-workflows/blob/main/server/workflow/workflow_server.go#L85-L89) is not called in the test, so the reflector is not started.